mithril.d.ts

```diff
@@ -149,18 +210,92 @@ declare module _mithril {
-		* @param args Arguments to optionally pass to the component.
-		* @return A component.
+		* @param component A mithril component. 
+		* @return A mithril component.
 		*
 		* @see m.render
 		* @see m.mount
 		* @see m
 		*/
-		component<T extends MithrilController>(
-			component: MithrilComponent<T>,
-			...args: any[]
-		): MithrilComponent<T>;
+		component<TController extends MithrilController>(
+			component: MithrilComponent<TController>
+		): MithrilComponent<TController>;
+
+		/**
+		* Initializes a parameterized component for use with m.render, m.mount, etc.
+		*
+		* @param component A parameterized component.
+		* @param arg1 First argument to bind to the component.
+		* @return A component bound with the supplied arguments.
+		*
+		* @see m.render
+		* @see m.mount
+		* @see m
+		*/
+		component<TController extends MithrilController, T1>(
+			component: ParameterizedMithrilComponent<TController, T1, void, void, void>,
+			arg1: T1
+		): MithrilComponent<TController>;
+
+		/**
+		* Initializes a parameterized component for use with m.render, m.mount, etc.
+		*
+		* @param component A parameterized component.
+		* @param arg1 First argument to bind to the component.
+		* @param arg2 Second argument to bind to the component.
+		* @return A component bound with the supplied arguments.
+		*
+		* @see m.render
+		* @see m.mount
+		* @see m
+		*/
+		component<TController extends MithrilController, T1, T2>(
+			component: ParameterizedMithrilComponent<TController, T1, T2, void, void>,
+			arg1: T1,
+			arg2: T2
+		): MithrilComponent<TController>;
+
+		/**
+		* Initializes a parameterized component for use with m.render, m.mount, etc.
+		*
+		* @param component A parameterized component.
+		* @param arg1 First argument to bind to the component.
+		* @param arg2 Second argument to bind to the component.
+		* @param arg3 Third argument to bind to the component.
+		* @return A component bound with the supplied arguments.
+		*
+		* @see m.render
+		* @see m.mount
+		* @see m
+		*/
+		component<TController extends MithrilController, T1, T2, T3>(
+			component: ParameterizedMithrilComponent<TController, T1, T2, T3, void>,
+			arg1: T1,
+			arg2: T2,
+			arg3: T3
+		): MithrilComponent<TController>;
+
+		/**
+		* Initializes a parameterized component for use with m.render, m.mount, etc.
+		*
+		* @param component A parameterized component.
+		* @param arg1 First argument to bind to the component.
+		* @param arg2 Second argument to bind to the component.
+		* @param arg3 Third argument to bind to the component.
+		* @param arg4 Fourth argument to bind to the component.
+		* @return A component bound with the supplied arguments.
+		*
+		* @see m.render
+		* @see m.mount
+		* @see m
+		*/
+		component<TController extends MithrilController, T1, T2, T3, T4>(
+			component: ParameterizedMithrilComponent<TController, T1, T2, T3, T4>,
+			arg1: T1,
+			arg2: T2,
+			arg3: T3,
+			arg4: T4
+		): MithrilComponent<TController>;
```

Comment:
Nit: keep the base `...args: any` type that was initially here.

This is probably better to separate into another PR.